src/param.c

```diff
@@ -142,6 +142,63 @@ stumpless_new_param( const char *name, const char *value ) {
   return result;
 }
 
+struct stumpless_param *
+stumpless_new_param_from_string( const char *string ) {
+
+  VALIDATE_ARG_NOT_NULL( string );
+
+  int i;
+  int name_len = 0, name_start = 0, value_len = 0, value_start = 0;
+
+  /* Check that the characters in 'name' are allowed. */
+  for (i = 0; string[i] != '='; i++) {
```

Comment:
what happens if there is no equals sign in the string? it seems like this loop will iterate past the end? Please add a test case for this possibility.

Reply:
If there is no equal sign, then we iterate until ```string[i]=='\0'``` in which case we will raise an error anyway (since ```\0``` is not an acceptable character for the name) and exit with NULL. Want me to add a clause to end the loop anyway?

Reply:
As long as you have a test case demonstrating that the case is handled correctly with the correct error, no changes are necessarily needed.